### DISCUSSION TITLE
Some work, some don't

### DISCUSSION THREAD

**Discussion by xBelo** [masterix21/green-pass#2] [Q&A]
Hi, I'm testing your code on one of my servers. 
Unfortunately, not all green passes are read: the ones that are tested in TestCase.php obviously work, others on the net too. 
But if I try my own and other people's gp, they are not recognized. In the error log I see this: "PHP Warning: zlib_decode (): data error"
Can you tell me how to solve this problem? Thanks

**Comment by masterix21** [comment]
Hi @xBelo, are you talking about real green pass or test green pass supplied by EU?

I tried a lots, but without issue. Would you mind helping me to dig inside the problem?

Thanks.

**Reply by xBelo** [reply]
Hi, no, I'm talking about the real green passes issued in this case by the Italian health ministry.
I tried about ten different green passes, all obviously issued by the ministry and none of them worked.
Only works with test ones.  
As I told you in the previous message, the error log gave zlib_decode (): data error.
I tried to dig into the code but i don't see anything wrong, so i ask you what it could be
thanks

**Reply by masterix21** [reply]
Sorry, but your information isn't enough to help you. If you can, please post a full exception stacktrace.

Thanks

**Reply by xBelo** [reply]
I get this:
```
PHP Warning:  zlib_decode(): data error in ...masterix21/green-pass/src/Services/Decoder.php on line 34
PHP Fatal error:  Uncaught InvalidArgumentException: Out of range. Expected: 1, read: 0. P Warning:  zlib_decode(): data error in .../masterix21/green-pass/src/Services/Decoder.php on line 34
PHP Fatal error:  Uncaught InvalidArgumentException: Out of range. Expected: 1, read: 0. in .../spomky-labs/cbor-php/src/StringStream.php:53\nStack trace:\n#0 .../spomky-labs/cbor-php/src/Decoder.php(48): CBOR\\StringStream->read()\n#1 .../spomky-labs/cbor-php/src/Decoder.php(43): CBOR\\Decoder->process()\n#2 .../masterix21/green-pass/src/Services/Decoder.php(58): CBOR\\Decoder->decode()\n#3 .../masterix21/green-pass/src/Services/Decoder.php(74): Masterix21\\GreenPass\\Services\\Decoder::cose()\n#4 .../index.php(43): Masterix21\\GreenPass\\Services\\Decoder::qrcode()\n#5 {main}\n  thrown in .../spomky-labs/cbor-php/src/StringStream.php on line 53
```

**Comment by nws-paolo** [comment]
hi,
we have the very same error while decoding raw data:

PHP Warning:  zlib_decode(): data error in /var/vhost/[...xyz...]/app/vendor/masterix21/green-pass/src/Services/Decoder.php on line 34

we are on debian 10, apache2, PHP 8.0.10

if we try to encode / decode simple strings zlib is working fine ...
what could be wrong?

**Comment by masterix21** [comment]
What kind of green pass are you reading? Vaccine? Test? Recovery?

I tried the library with all my friends' green-passes, and it works: I have no idea. Add a PR to fix the issue, or send me the green pass that causes the problem.

Thanks

**Reply by nws-paolo** [reply]
hi, and thank you for your reply.
we discoverd a 2D barcode reader was faulty and it was returning unpredictable wrong strings.
Our bad, we didn't realize this before. Thanks once again for your support.
paolo

**Reply by masterix21** [reply]
It sounds good :) I'm thrilled to read your feedback. Thanks!

**Comment by masterix21** [comment]
@xBelo, please check if your reader includes carriage characters (or something else) at the end of the string. Thanks
